src/model/core/fuel.jl

```diff
@@ -0,0 +1,152 @@
+(1). Use a constant heat rate, regardless of the minimum load or maximum load; and 
+(2). use the "PiecewiseFuelUsage" options to model the fuel consumption via piecewise linear 
+approximation. 
+By using that option, users could represent higher heat rate when generators are running at 
+minimum load, and lower heatrate when generators are running at maximum load.
+
+(1). Constant heat rate. 
+The fuel consumption for power generation $vFuel_{y,t}$ is determined by power generation 
+($vP_{y,t}$) mutiplied by the corresponding heat rate ($Heat\_Rate_y$). 
+The fuel costs for power generation and start fuel for a plant $y$ at time $t$, 
+denoted by $eCFuelOut_{y,t}$ and $eFuelStart$, is determined by fuel consumption ($vFuel_{y,t}$ 
+and $eStartFuel$) multiplied by the fuel costs (\$/MMBTU)
+(2). Piecewise linear approximation
+In the first formulations, thermal generators are expected to have the same fuel consumption 
+per generating 1 MWh electricity,
+regardless of minimum load or full load. However, thermal generators tend to have decreased 
+efficiency when operating at part load, leading to higher fuel consumption per generating 
+the same amount of electricity. 
+To have more precise representation of fuel consumption at part load, 
+the piecewise-linear fitting of heat input can be introduced. 
+
+```math
+\begin{aligned}
+vFuel_{y,t} >= vP_{y,t} * h_{y,x} + U_{g,t}* f_{y,x}
+\hspace{1cm} \forall y \in G, \forall t \in T, \forall x \in X
+\end{aligned}
+```
+Where $h_{y,x}$ represents slope a thermal generator $y$ in segment $x$ [MMBTU/MWh],
+and $f_{y,x}$ represents intercept (MMBTU) of a thermal generator $y$ in segment $x$ [MMBUT],
+and $U_{y,t}$ represents the commit status of a thermal generator $y$ at time $t$.
+In "Generators_data.csv", users need to specify the number of segments in a column called 
+PWFU_NUM_SEGMENTS, then provide corresponding Slopei and Intercepti based on the PWFU_NUM_SEGMENTS 
+(i = PWFU_NUM_SEGMENTS).
+
+Since fuel consumption (and fuel costs) is postive,
+the optimization will optimize the fuel consumption by enforcing the inequity to equal to 
+the highest piecewise segment.Only one segment is active at a time for any value of vP. 
+When the power output is zero, the commitment variable $U_{g,t}$ will bring the intercept 
+to be zero such that the fuel consumption is zero when thermal units are offline.
+
+In order to run piecewise fuel consumption module,
+the unit commitment must be turned on, and users should provide Slope and 
+Intercept for at least one segment. 
+"""
+
+function fuel!(EP::Model, inputs::Dict, setup::Dict)
+    println("Fuel Module")
+    dfGen = inputs["dfGen"]
+    T = inputs["T"]     # Number of time steps (hours)
+    Z = inputs["Z"]     # Number of zones
+    G = inputs["G"]     
+    FUEL = inputs["FUEL"]
+    THERM_COMMIT = inputs["THERM_COMMIT"]
+    # exclude the "None" from fuels
+    fuels = inputs["fuels"]
```

Comment:
It's generally a bit confusing that now we have `FUEL` and `fuels` as sets. The lowercase `fuels` already exists, and is a `Vector{String}` right? Could `FUEL` be called `HAS_FUEL` or something else similar?

Reply:
yep fuels is a vector{string}. Goood suggestions! I will change it to HAS_FUEL